lib/util/exec/docker/index.ts

```diff
@@ -68,5 +74,12 @@ export function dockerCmd(
 
   result.push(cmd);
 
-  return result.join(' ');
+  const mainCommand = result.join(' ');
+
+  if (prefetchedDockerImages.has(taggedImage)) {
+    return [mainCommand];
+  }
+
+  prefetchedDockerImages.add(taggedImage);
+  return [`docker pull ${taggedImage}`, mainCommand];
```

Comment:
No, let's pull it using a separate child_process. We don't want it to be counted as part of the time taken to exec

Reply:
Let's merge #5207, then I'll deconflict & update it

Reply:
ok. I'm having test case failures for some reason, hope to resolve soon

Reply:
Can be fixed by `join('...')` from `upath`, if it's Windows CI failure

Reply:
I skipped it, updated this branch from master